### DISCUSSION TITLE
One suggestion for a future version

### DISCUSSION THREAD

**Discussion by dcousin3** [jspsych/jsPsych#937] [Ideas]
Hello,

jsPsych is really a great environment to develop experiments, thanks for developping and maintaining it! 

One feature that I find very usefull is the trigger `on_finish`. It is very convenient to run short functions (without using function-call). However, if I understand correctly, this triggers an action performed at the end of all the objects in the timeline. It would be usefull IMO to have another trigger, say `on_timeline_finish`, that would be triggered only once when the complete timeline is exhausted. That could be convenient for example if the timeline is a sequence of events in which a few responses are obtained, to gather all the responses in a single vector. With `on_finish`, it would gather all the responses individually, not collectively.

Keep the good work and congratulations for your well-deserved Mozilla award!
Denis Cousineau.

**Comment by jodeleeuw** [comment]
Thanks @dcousin3 ! We totally agree about this feature. In fact, it's on our todo list for version 7.0: #585
